### ISSUE TITLE
Getting equivalence between torchvision and image transforms when normalizing without rescaling

### ISSUE THREAD

**Issue by NielsRogge** [huggingface/transformers#26133]
### System Info

Torchvision version 0.14.0
Transformers version 4.27.0.dev0

### Reproduction

Trying to have equivalent performance for the following (relevant for #26092):

```
from transformers.image_transforms import normalize

from PIL import Image
import requests
import numpy as np
import torch

from torchvision import transforms


url = "https://dl.fbaipublicfiles.com/dinov2/images/example.jpg"
image = Image.open(requests.get(url, stream=True).raw)

def make_depth_transform() -> transforms.Compose:
    return transforms.Compose([
        transforms.ToTensor(),
        lambda x: 255.0 * x[:3], # Discard alpha component and scale by 255
        transforms.Normalize(
            mean=(123.675, 116.28, 103.53),
            std=(58.395, 57.12, 57.375),
        ),
    ])

# prepare image using torchvision
transform = make_depth_transform()
original_pixel_values = transform(image)

# prepare image using Transformers' image transforms
pixel_values = normalize(np.array(image), mean=(123.675, 116.28, 103.53), std=(58.395, 57.12, 57.375))
pixel_values = np.moveaxis(pixel_values, -1, 0)
pixel_values = torch.from_numpy(pixel_values).float()

assert original_pixel_values.shape == pixel_values.shape
assert torch.allclose(original_pixel_values, pixel_values)
```
The assertion above currently fails.

### Expected behavior

I would like to have equivalent behaviour between torchvision and our image transforms when normalizing without rescaling.

cc @amyeroberts

**Comment by amyeroberts** [comment]
@NielsRogge This is happening because the dtype of the images when calling the torch and image_transforms `normalize` aren't the same. If you cast the dtype to float32, then you'll get the equivalent behaviour: 

```py
pixel_values = normalize(np.array(image).astype(np.float32), mean=(123.675, 116.28, 103.53), std=(58.395, 57.12, 57.375))
```

This is because `ToTensor()` is doing three things: casting to float32, rescaling, converting to a torch tensor.

**Comment by NielsRogge** [comment]
Thanks a lot, this was what I needed!
